lib/failure.sh

```diff
@@ -27,6 +27,20 @@ fail_invalid_package_json() {
   fi
 }
 
+fail_dot_heroku_node() {
+  if [ -f "${1:-}/.heroku/node" ]; then
```

Comment:
It won't cause an error since you `mkdir -p`, but the question is, do we want users' unknown stuff in there?